controllers/src/main/scala/cromwell/pipeline/controller/AkkaHttpClient.scala

```diff
@@ -4,67 +4,113 @@ import akka.actor.ActorSystem
 
-  override def get(url: String, params: Map[String, String] = Map(), headers: Map[String, String] = Map())(
-    implicit ec: ExecutionContext
-  ): Future[Response] = {
+  override def get[B](url: String, params: Map[String, String] = Map(), headers: Map[String, String] = Map())(
+    implicit ec: ExecutionContext,
+    f: Format[B]
+  ): Future[Response[B]] = {
     val futureResponse: Future[HttpResponse] = Http().singleRequest(
       HttpRequest(method = HttpMethods.GET, uri = Uri(url).withQuery(Query(params))).withHeaders(parseHeaders(headers))
     )
     responsify(futureResponse)
   }
 
-  private def responsify(futureResponse: Future[HttpResponse])(implicit ec: ExecutionContext): Future[Response] =
+  private def responsify[B](
+    futureResponse: Future[HttpResponse]
+  )(implicit ec: ExecutionContext, f: Format[B]): Future[Response[B]] =
     futureResponse.flatMap { response =>
-      response.entity.toStrict(expirationTime).map { body =>
-        Response(
+      response.status match {
+        case StatusCodes.Created    => positiveResponse(response)(ec, f)
+        case StatusCodes.Accepted   => positiveResponse(response)(ec, f)
+        case StatusCodes.BadRequest => negativeResponse(response)(ec)
+        case StatusCodes.NotFound   => negativeResponse(response)(ec)
+        case _                      => negativeResponse(response)(ec)
+      }
+    }
+
+  private def positiveResponse[B](
+    response: HttpResponse
+  )(implicit ec: ExecutionContext, f: Format[B]): Future[Response[B]] =
+    if (response.headers.contains(httpContentTypeJson)) {
+      getResponseBody[B](response)(ec, f).map { responseBody =>
+        Response[B](
           response.status.intValue(),
-          body.data.utf8String,
-          response.headers.foldLeft(Map[String, List[String]]()) {
-            case (acc, header) =>
-              acc + (header.name() -> (header.value() :: acc.getOrElse(header.name(), Nil)))
-          }
+          PositiveResponseBody(responseBody.get),
+          parseHeaders(response)
         )
       }
+    } else {
+      Future.failed(ParseJsonException("Could not parse body cause it is not a json"))
+    }
+
+  private def negativeResponse[B](response: HttpResponse)(implicit ec: ExecutionContext): Future[Response[B]] =
+    response.entity.toStrict(expirationTime).map { body =>
+      val pb = parseResponseBody(body)
+      Response[B](
+        response.status.intValue(),
+        NegativeResponseBody(pb.toString()),
+        parseHeaders(response)
+      )
+    }
+
+  private def parseHeaders(response: HttpResponse): Map[String, Seq[String]] =
+    response.headers.foldLeft(Map[String, List[String]]()) {
+      case (acc, header) =>
+        acc + (header.name() -> (header.value() :: acc.getOrElse(header.name(), Nil)))
+    }
+
+  private def getResponseBody[B](
+    response: HttpResponse
+  )(implicit ec: ExecutionContext, f: Format[B]): Future[JsResult[B]] =
```

Comment:
For this method Reads instead of Format is enough to work correctly

Reply:
that's right, I suppose. Done